components/filter-field/src/filter-field.scss

```diff
@@ -110,6 +110,11 @@ button.dt-filter-field-clear-all-button {
   margin-left: 8px;
 }
 
+button.dt-filter-field-clear-all-button-hidden {
```

Comment:
why we have to hide it? Can we just `ngIf` it, please explain it to me :)

Reply:
As seen in the description of the PR, if we *ngIf the button, it will interfere with how the filterfield uses the focusmonitor to show the button or not. The button would be removed on mouseDown, would immediately loose focus (because it is being removed), and be shown again, without the click handler ever firing.

> The clearAll button rendering depended on the filterfield not being
focussed. When the clearAll button was rendered and was clicked, the
cdkFocusMonitor kicked in, set the focus and removed the clearAll button
again. As a result the clearAll button lost its focus, the document
focus was set back to the body and the clearAll was rendered again. The
click event on the clearAll was never fired.
We no longer remove the clearAll from the DOM, but hide it via css
classes. This makes it not lose focus, and results in a more consistent
behavior.